### ISSUE TITLE
[openSUSE Tumbleweed] rootless: x509 certificates may not load properly on some systems

### ISSUE THREAD

**Issue by terencehonles** [moby/moby#42080]
**Description**

I opened up the upstream issue https://github.com/rootless-containers/rootlesskit/issues/225 but I had tried to install docker rootless awhile ago at this point. I tried debugging why I was getting x509 errors, but I didn't have time to look into it until now. I am mainly opening up this issue in order to potentially update the documentation and/or have this is something a user might find when looking into why rootless is not working for them.

**Steps to reproduce the issue:**

1. Have a system with symlinks inside one of /etc or /run as described in https://github.com/rootless-containers/rootlesskit/issues/225 (This is the case for my current installation of openSUSE Tumbleweed)
2. Install docker rootless as per the documentation
3. Try to `docker pull` any image

**Describe the results you received:**

```
Error response from daemon: Get https://registry-1.docker.io/v2/: x509: certificate signed by unknown authority
```

**Describe the results you expected:**

A successful pull

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           20.10.3
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        48d30b5
 Built:             Fri Jan 29 14:28:23 2021
 OS/Arch:           linux/amd64
 Context:           rootless
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.3
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       46229ca
  Built:            Fri Jan 29 14:31:57 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker info`:**

```
Client:
 Context:    rootless
 Debug Mode: false

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 1
 Server Version: 20.10.3
 Storage Driver: btrfs
  Build Version: Btrfs v4.20.1 
  Library Version: 102
 Logging Driver: json-file
 Cgroup Driver: none
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  rootless
 Kernel Version: 5.10.16-1-default
 Operating System: openSUSE Tumbleweed
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.51GiB
 Name: osprey
 ID: 35CV:EJFU:FWZ2:S2IE:ZY3P:P2NA:FYJV:WR4C:PVUU:QMNE:UYGA:BYVE
 Docker Root Dir: /home/terence/.local/share/docker
 Debug Mode: true
  File Descriptors: 23
  Goroutines: 40
  System Time: 2021-02-25T13:25:44.573426464-08:00
  EventsListeners: 0
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

WARNING: Running in rootless-mode without cgroups. To enable cgroups in rootless-mode, you need to boot the system in cgroup v2 mode.
```

**Comment by AkihiroSuda** [comment]
Workaround:
```patch
--- dockerd-rootless.sh.OLD     2021-02-26 07:46:56.137467327 +0100
+++ dockerd-rootless.sh 2021-02-26 07:51:09.832257325 +0100
@@ -95,5 +95,12 @@
        # remove the symlinks for the existing files in the parent namespace if any,
        # so that we can create our own files in our mount namespace.
        rm -f /run/docker /run/containerd /run/xtables.lock
+
+       # Workaround for https://github.com/rootless-containers/rootlesskit/issues/225
+       realpath_etc_ssl=$(realpath /etc/ssl)
+       rm -f /etc/ssl
+       mkdir /etc/ssl
+       mount --rbind ${realpath_etc_ssl} /etc/ssl
+
        exec dockerd $@
 fi
```

**Comment by AkihiroSuda** [comment]
Resolved in Docker 20.10.8
https://github.com/moby/moby/pull/42462
